src/psyclone/tests/psyir/transformations/kernel_transformation_test.py

```diff
@@ -266,6 +267,121 @@ def test_new_same_kern_single(kernel_outputdir, monkeypatch):
+        rtrans.validate(kern)
+    assert ("Failed to create PSyIR for kernel 'testkern_code'. Cannot "
+            "transform such a kernel." in str(err.value))
+
+
+def test_accroutinetrans_validate_no_import(fortran_reader):
+    '''
+    Test the validate() method of ACCRoutineTrans rejects a kernel that
+    accesses imported data unless that data is known to be a compile-
+    time constant.
+
+    '''
+    code = '''\
+module my_mod
+  use other_mod, only: some_data
+contains
+  subroutine my_sub(arg)
+    integer :: arg
+    arg = arg + some_data
+  end subroutine my_sub
+end module my_mod'''
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Routine)[0]
+    rtrans = ACCRoutineTrans()
+    with pytest.raises(TransformationError) as err:
+        rtrans.validate(routine)
+    assert ("Transformation Error: routine 'my_sub' accesses the symbol "
+            "'some_data: Symbol<Import(container='other_mod')>' which is "
+            "imported. If this symbol represents data "
+            "then it must first be converted to a routine argument using the "
+            "KernelImportsToArguments transformation." in str(err.value))
+    # Specialise the imported symbol and make it constant.
+    sym = psyir.children[0].symbol_table.lookup("some_data")
+    sym.specialise(DataSymbol, datatype=INTEGER_TYPE, is_constant=True)
+    # Validation should now pass.
+    rtrans.validate(routine)
+
+
+def test_accroutinetrans_validate_no_import_cblock(fortran_reader):
+    '''
+    Test the validate() method of ACCRoutineTrans rejects a kernel that
+    accesses imported data when that access is within a CodeBlock.
+
+    '''
+    code = '''\
+module my_mod
+  use other_mod, only: some_data
+contains
+  subroutine my_sub(arg)
+    integer :: arg
+    write(*,*) arg + some_data
+  end subroutine my_sub
+end module my_mod'''
+    psyir = fortran_reader.psyir_from_source(code)
+    routine = psyir.walk(Routine)[0]
+    rtrans = ACCRoutineTrans()
+    with pytest.raises(TransformationError) as err:
+        rtrans.validate(routine)
+    assert ("Transformation Error: routine 'my_sub' accesses the symbol "
+            "'some_data' within a CodeBlock and this symbol is imported. "
+            "'ACC routine' cannot be added to such a routine."
+            in str(err.value))
+
+
+def test_accroutinetrans_validate_no_call():
+    '''
+    Test the validate() method of ACCRoutineTrans rejects a kernel that calls
+    another routine.
+
+    '''
+    psy, invoke = get_invoke("1.15_invoke_kern_with_call.f90", api="dynamo0.3",
+                             idx=0)
+    sched = invoke.schedule
+    kernel = sched.coded_kernels()[0]
+    rtrans = ACCRoutineTrans()
+    with pytest.raises(TransformationError) as err:
+        rtrans.validate(kernel)
+    assert ("Kernel 'testkern_with_call_code' calls another routine "
+            "('call xyz2llr(coord(1), coord(2), coord(3), lon, lat, radius)') "
```

Comment:
I would say that the outer parenthesis in this error message are not needed if we already have the single quotes.

Reply:
So would I.